test/AsmJs/uint.js

```diff
@@ -12,10 +12,9 @@ let m = function (stdlib, foreign, heap) {
   }
   return f;
 }({}, {}, asmHeap);
-print(m(4294967295));
-print(m(4294967295));
 
-m = function (stdlib, foreign, heap) {
+
+m2 = function (stdlib, foreign, heap) {
```

Comment:
no, it should be deterministic. my guess is it's because m2 is now becoming a property on global object rather than a var. if you change to `let m2 = ...`, I suspect behavior will revert back.